openmc/data/multipole.py

```diff
@@ -129,6 +143,608 @@ def _broaden_wmp_polynomials(E, dopp, n):
+
+    # detect peaks (resonances) and determine VF order search range
+    peaks, _ = find_peaks(ce_xs[0] + ce_xs[1])
+    n_peaks = peaks.size
+    if orders is not None:
+        # make sure orders are even integers
+        orders = list(set([int(i/2)*2 for i in orders if i >= 2]))
+    else:
+        lowest_order = max(2, 2*n_peaks)
+        highest_order = max(200, 4*n_peaks)
+        orders = list(range(lowest_order, highest_order + 1, 2))
+
+    if log:
+        print("Found {} peaks".format(n_peaks))
+        print("Fitting orders from {} to {}".format(orders[0], orders[-1]))
+
+    # perform VF with increasing orders
+    found_ideal = False
+    n_discarded = 0  # for accelation, number of discarded searches
+    best_quality = best_ratio = -np.inf
+    for i, order in enumerate(orders):
+        if log:
+            print("Order={}({}/{})".format(order, i, len(orders)))
+        # initial guessed poles
+        poles_r = np.linspace(s[0], s[-1], order//2)
+        poles = poles_r + poles_r*0.01j
+        poles = np.sort(np.append(poles, np.conj(poles)))
+
+        found_better = False
+        # fitting iteration
+        for i_vf in range(n_vf_iter):
+            if log >= DETAILED_LOGGING:
+                print("VF iteration {}/{}".format(i_vf + 1, n_vf_iter))
+
+            # call vf
+            poles, residues, cf, f_fit, rms = vf.vectfit(f, s, poles, weight)
+
+            # convert real pole to conjugate pairs
+            n_real_poles = 0
+            new_poles = []
+            for p in poles:
+                p_r, p_i = np.real(p), np.imag(p)
+                if (s[0] <= p_r <= s[-1]) and p_i == 0.:
+                    new_poles += [p_r+p_r*0.01j, p_r-p_r*0.01j]
+                    n_real_poles += 1
+                else:
+                    new_poles += [p]
+            new_poles = np.array(new_poles)
+            # re-calculate residues if poles changed
+            if n_real_poles > 0:
+                if log >= DETAILED_LOGGING:
+                    print("  # real poles: {}".format(n_real_poles))
+                new_poles, residues, cf, f_fit, rms = \
+                      vf.vectfit(f, s, new_poles, weight, skip_pole=True)
+
+            # assess the result on test grid
+            test_xs = vf.evaluate(test_s, new_poles, residues) / test_energy
+            abserr = np.abs(test_xs - test_xs_ref)
+            with np.errstate(invalid='ignore', divide='ignore'):
+                relerr = abserr / test_xs_ref
+                if np.any(np.isnan(abserr)):
+                    maxre, ratio, ratio2 = np.inf, -np.inf, -np.inf
+                elif np.all(abserr <= atol):
+                    maxre, ratio, ratio2 = 0., 1., 1.
+                else:
+                    maxre = np.max(relerr[abserr > atol])
+                    ratio = np.sum((relerr < rtol) | (abserr < atol)) / relerr.size
+                    ratio2 = np.sum((relerr < 10*rtol) | (abserr < atol)) / relerr.size
```

Comment:
Should this block really be under `with np.errstate`? I thought only the `relerr = abserr / test_xs_ref` line would cause warnings.

Reply:
`relerr` contains `invalid value` (`nan`) so operations on `relerr` in the following lines, like `(relerr < rtol)` will cause warnings as well. 
The following piece of code will give two times of `RuntimeWarning: invalid value encountered`:
```
import numpy as np
a = b = np.array([[1,2,3.],[0., 0., 0.]])
c = b/a
c < 0.001
```